### PR TITLE
Preserve indentation for first line of <pre> content

### PR DESCRIPTION
**Pull request by jeremy-jameson**
I encountered an issue where indentation on the first line of code within a `<pre>` element is removed when converting to Markdown.

For example, consider the first block of code from one of my blog posts: [Unit tests for filtering errors in ELMAH](https://www.technologytoolbox.com/blog/jjameson/archive/2012/02/28/unit-tests-for-filtering-errors-in-elmah.aspx)

```C#
        [TestMethod()]
        public void TypeAssertionTest001()
        {
        ...
```

ReverseMarkdown v3.18.0 converts this to

```C#
[TestMethod()]
        public void TypeAssertionTest001()
        {
        ...
```

In other words, ReverseMarkdown is not preserving the "preformatted" content.

The fix is simple -- remove line 18 in **Pre.cs** (which trims the content). However, this requires updating a number of existing tests to reflect the fact that _all_ whitespace in `<pre>` content is significant.

### CODE REVIEW

**Comment by jeremy-jameson** [comment]
I accidentally "polluted" this pull request when I checked in additional changes in my fork of ReverseMarkdown. I will close this pull request and open a new one with the changes isolated on a separate branch.

Note that I am in the process of converting my blog to Markdown and there are some additional changes I am going to make to my fork of ReverseMarkdown. Whether you choose to merge those changes into ReverseMarkdown is completely up to you, but I will try to ensure each change can be isolated into its own pull request.
